### ISSUE TITLE
Publications category menu should have an accessible label

### ISSUE THREAD

**Issue by eatyourgreens** [zooniverse/front-end-monorepo#4765]
## Package
app-content-pages
## Describe the bug
The sidebar navigation on the publications page is missing an accessible label for the category menu.
```html
<nav class="StyledBox-sc-13pk1d4-0 ezNNQt">
  <ul class="StyledBox-sc-13pk1d4-0 bjWreY">
    <li class="sc-5b224d84-0 LmnZY"><a class="StyledButton-sc-323bzc-0 eerEGI sc-5b224d84-1 jTtzKn" href="/about/publications">Show all</a></li>
    <li class="sc-5b224d84-0 LmnZY"><a class="StyledButton-sc-323bzc-0 fxCSHo sc-5b224d84-1 kCWXRm" href="/about/publications#space">Space</a></li>
    <li class="sc-5b224d84-0 LmnZY"><a class="StyledButton-sc-323bzc-0 fxCSHo sc-5b224d84-1 kCWXRm" href="/about/publications#physics">Physics</a></li>
    <li class="sc-5b224d84-0 LmnZY"><a class="StyledButton-sc-323bzc-0 fxCSHo sc-5b224d84-1 kCWXRm" href="/about/publications#climate">Climate</a></li>
    <li class="sc-5b224d84-0 LmnZY"><a class="StyledButton-sc-323bzc-0 fxCSHo sc-5b224d84-1 kCWXRm" href="/about/publications#humanities">Humanities</a></li>
    <li class="sc-5b224d84-0 LmnZY"><a class="StyledButton-sc-323bzc-0 fxCSHo sc-5b224d84-1 kCWXRm" href="/about/publications#nature">Nature</a></li>
    <li class="sc-5b224d84-0 LmnZY"><a class="StyledButton-sc-323bzc-0 fxCSHo sc-5b224d84-1 kCWXRm" href="/about/publications#medicine">Medicine</a></li>
    <li class="sc-5b224d84-0 LmnZY"><a class="StyledButton-sc-323bzc-0 fxCSHo sc-5b224d84-1 kCWXRm" href="/about/publications#social-science">Social Science</a></li>
    <li class="sc-5b224d84-0 LmnZY"><a class="StyledButton-sc-323bzc-0 fxCSHo sc-5b224d84-1 kCWXRm" href="/about/publications#language">Language</a></li>
    <li class="sc-5b224d84-0 LmnZY"><a class="StyledButton-sc-323bzc-0 fxCSHo sc-5b224d84-1 kCWXRm" href="/about/publications#meta">Meta</a></li>
    <li class="sc-5b224d84-0 LmnZY"><a class="StyledButton-sc-323bzc-0 fxCSHo sc-5b224d84-1 kCWXRm" href="/about/publications#data">Data</a></li>
  </ul>
</nav>
```

## To Reproduce
Inspect the sidebar menu here:
https://frontend.preview.zooniverse.org/about/publications

## Expected behavior
Since https://github.com/zooniverse/front-end-monorepo/pull/4673, the menu should be labelled something like 'Filter by category'.
